torch/distributed/_tensor/ops/basic_strategy.py

```diff
@@ -0,0 +1,176 @@
+import itertools
+from dataclasses import dataclass
+
+from typing import List, Tuple
+
+from torch.distributed._tensor.device_mesh import DeviceMesh
+from torch.distributed._tensor.op_schema import PlacementStrategy, StrategyList
+from torch.distributed._tensor.placement_types import (
+    _Partial,
+    DTensorSpec,
+    Placement,
+    Replicate,
+    Shard,
+)
+
+
+@dataclass
+class EinsumDims:
+    contracting_dims: List[str]
+    batch_dims: List[str]
+    lhs_free_dims: List[str]
```

Comment:
These are the dims that appear in the lhs and output but not the rhs. `free` feels like confusing name because it suggests "free variable" meaning it's not bound in the lhs. Maybe `lhs_only` or `lhs_output_only`.

Reply:
make sense, changed.